### ISSUE TITLE
Improve literal min/max checking.

### ISSUE THREAD

**Issue by jacqueswww** [ethereum/vyper#844]
### What's your issue about?

From gitter:

We do fold the constants in python. That's why 2**256 - 1 throws out of range error
basically the AST parses 2 ** 256 to POW(2, 256), then we pick it up becomes a new literal. And then that new literal fails - because that literal is out of bounds as per our definition.
The problem I had was, I couldn't figure out how to know - this is the final stage literal.
But I assume now that I think about it some more, I could just move those limit checks to the assignment/comparison operator cases.
basically literals can be huge, but the moment they get used we do the checks.

### How can it be fixed?

Fill this in if you know how to fix it.

#### Cute Animal Picture
![](https://timedotcom.files.wordpress.com/2017/01/cute-animal-tweet-off-zoo.jpg?quality=85)

**Comment by jacqueswww** [comment]
Starting on this.
